sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -73,9 +75,25 @@ def truncate(self, json_event):
         else:
             return json_event
 
+    def send_truncated_msg(self, db_name, time_spent):
+        event = {**self._base_event, "metadata": [], "timestamp": time.time() * 1000}
+        db_info = {}
+        if db_name not in self.db_to_schemas:
+            db_info = self.db_info[db_name]
+        else:
+            db_info = {"name": db_name}
+        db_info["truncated"] = "Truncated after fetching {} columns, elapsed time is {}s".format(
+            self._total_columns_sent, time_spent
+        )
+        event["metadata"] = [{**(db_info)}]
```

Comment:
Do not set this on the `metadata` field that is suppose to be for legit data, which AFAIK this field could still contain data? It would just not contain _all_ of the schema data, right? 

This is going to cause deserialization errors in the backend. Here is an example: https://github.com/DataDog/dd-go/blob/prod/database-monitoring/apps/dbm-metadata-processor/decoder/fixtures/pg_databases.json#L11

I would add a new field to the payload like `collection_errors` where you can store this info and then add some code to parse it in the backend:

```json
collection_errors: {
 error: truncated
 msg: "Truncated after fetching {} columns, elapsed time is {}s".format(
            self._total_columns_sent, time_spent
        )
}
```

Reply:
Backend code will work as I changed it. But I see your point, ok will change

Reply:
done , @jmeunier28 pls take another look.
(working on the backend change now)